filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobFileSystemManager.scala

```diff
@@ -96,19 +96,31 @@ case class WSMBlobTokenGenerator(
   container: BlobContainerName,
   endpoint: EndpointURL,
   workspaceId: WorkspaceId,
-  wsmClient: WorkspaceManagerApiClientProvider) extends BlobTokenGenerator {
-
-  def generateAccessToken: Try[AzureSasCredential] = Try {
-    val token = wsmClient.getControlledAzureResourceApi.createAzureStorageContainerSasToken(
-      UUID.fromString(workspaceId.value),
-      UUID.fromString("00001111-2222-3333-aaaa-bbbbccccdddd"),
-      null,
-      null,
-      null,
-      null
-    ).getToken // TODO `null` items may be required, investigate in WX-696
-
-    new AzureSasCredential(token) // TODO Does `signature` actually mean token? save for WX-696
+  containerResourceId: ContainerResourceId,
+  wsmClient: WorkspaceManagerApiClientProvider,
+  defaultB2cToken: Option[String]) extends BlobTokenGenerator {
+
+  def generateAccessToken: Try[AzureSasCredential] = {
+    val controlledAzureResourceApi: Try[ControlledAzureResourceApi] = defaultB2cToken match {
+      case Some(defaultToken) => Success(wsmClient.getControlledAzureResourceApi(defaultToken))
+      case None => AzureCredentials(None).getAccessToken.toOption match {
+        case Some(localAzureCredential) => Success(wsmClient.getControlledAzureResourceApi(localAzureCredential))
+        case None => Failure(new RuntimeException("B2C token not found"))
+      }
+    }
+    val token = controlledAzureResourceApi.flatMap((api: ControlledAzureResourceApi) => {
+      Try {
+        api.createAzureStorageContainerSasToken(
+          UUID.fromString(workspaceId.value),
+          UUID.fromString(containerResourceId.value),
```

Comment:
Now that we have this config set up, I think it would be better to transform these strings to UUID during config parsing. That way we fail fast if the values aren't valid.